Formula/glib-utils.rb

```diff
@@ -0,0 +1,106 @@
+class GlibUtils < Formula
+  include Language::Python::Shebang
+
+  desc "Python utilities for GLib"
+  homepage "https://developer.gnome.org/glib/"
+  url "https://download.gnome.org/sources/glib/2.72/glib-2.72.2.tar.xz"
+  sha256 "78d599a133dba7fe2036dfa8db8fb6131ab9642783fc9578b07a20995252d2de"
+  license "LGPL-2.1-or-later"
+
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "glib"
+  depends_on "python@3.10"
+
+  def install
+    # TODO: This is a workaround for `brew audit --new-formula`.
+    #       Use `patch` rather than `inreplace` (see also `glib`).
+    # replace several hardcoded paths with homebrew counterparts
+    inreplace "gio/xdgmime/xdgmime.c",
+              'xdg_data_dirs = "/usr/local/share/:/usr/share/";',
+              "xdg_data_dirs = \"#{HOMEBREW_PREFIX}/share/:/usr/local/share/:/usr/share/\";"
+    inreplace "glib/gutils.c",
+              'data_dirs = "/usr/local/share/:/usr/share/";',
+              "data_dirs = \"#{HOMEBREW_PREFIX}/share/:/usr/local/share/:/usr/share/\";"
+
+    # Point the headers and libraries to `glib`.
+    # The headers and libraries will be removed later because they are provided by `glib`.
+    glib = Formula["glib"]
+    args = std_meson_args.delete_if do |arg|
+      arg.start_with?("--includedir=", "--libdir=")
+    end
+    args += %W[
+      --includedir=#{glib.opt_include}
+      --libdir=#{glib.opt_lib}
+    ]
+
+    # Disable dtrace; see https://trac.macports.org/ticket/30413
+    # and https://gitlab.gnome.org/GNOME/glib/-/issues/653
+    args += %W[
+      --default-library=both
+      --localstatedir=#{var}
+      -Diconv=auto
+      -Dgio_module_dir=#{HOMEBREW_PREFIX}/lib/gio/modules
+      -Dbsymbolic_functions=false
+      -Ddtrace=false
+    ]
+
+    mkdir "build" do
+      system "meson", *args, ".."
+      system "ninja", "-v"
+
+      # Skip files already in glib
+      Formula["meson"].opt_libexec.cd do
+        system "bin/python3", "-c", pyscript
+      end
+      system "ninja", "install", "-v"
+    end
+
+    # Delete non python files because they are provided by `glib`
+    dummy_python_shebang_rewrite_info = python_shebang_rewrite_info("#{HOMEBREW_PREFIX}/bin/python3")
+    prefix.find do |f|
+      next unless f.file?
+      next if f.extname == ".py"
+      next if dummy_python_shebang_rewrite_info.regex.match?(f.read(dummy_python_shebang_rewrite_info.max_length))
+
+      f.unlink
+    end
+
+    # Delete empty directories
+    prefix.children
```

Comment:
```suggestion
    prefix.find
```
Sorry, this should be `#find`. `#children` only goes one level down.

Reply:
I revert to the previous implementation here.

```ruby
dirs.select(&:empty?).map(&:rmdir)
```

and

```ruby
dirs.each { |d| d.rmdir if d.empty? }
```

are not the same.

------

```ruby
# Delete empty directories
# Note: We need to traversal the directories in reverse order (i.e. deepest first).
#       Also, we should put checking emptiness and deletion in a single loop.
#       `dirs.select(&:empty?).map(&:rmdir)` will not work because it will not delete
#       directories that only contain empty directories.
prefix.glob("**/").reverse_each { |d| d.rmdir if d.empty? }
```